### PR TITLE
Add EIP-6617: Add a new bit based permission proposal

### PR DESCRIPTION
**Pull request by chiro-hiro**

### CODE REVIEW

**Comment by chiro-hiro** [comment]
Hi @Pandapip1,

We add a new **Bit Based Permission** proposal as your request. I and @vdusart will update EIP-6366 as well.

**Comment by abcoathup** [comment]
The PR doesn't have a `c-new` label?

File: EIPS/eip-bit_based_permission.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+---
+
+## Abstract
+
+This EIP offers a standard for building a bit-based permission and role system. Each permission is represented by a single bit in `uint256` from which we can defined up to `256` permissions and `2²⁵⁶` roles. We are able to specify the importance of permission based on the order of the bits.
+
+## Motivation
+
+Currently permission and access control are done using `string` or `keccak256(string)` comparaisons (see [ERC-5982](./eip-5982.md)).
+By using bitwise and bitmask operations to determine access rights, we gain in efficiency, flexibility and more.
```
Comment (Pandapip1) [inline EIPS/eip-bit_based_permission.md:20]:
```suggestion
```
This reads like Rationale.

File: EIPS/eip-bit_based_permission.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+discussions-to: https://ethereum-magicians.org/t/bit-based-permission/13065
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-02-27
+---
+
+## Abstract
+
+This EIP offers a standard for building a bit-based permission and role system. Each permission is represented by a single bit in `uint256` from which we can defined up to `256` permissions and `2²⁵⁶` roles. We are able to specify the importance of permission based on the order of the bits.
```
Comment (Pandapip1) [inline EIPS/eip-bit_based_permission.md:15]:
```suggestion
This EIP offers a standard for building a bit-based permission and role system. Each permission is represented by a single bit. By using an `uint256`, up to $256$ permissions and $2^{256}$ roles can be defined. We are able to specify the importance of permission based on the order of the bits.
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

File: EIPS/eip-bit_based_permission.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+### Ordering permissions by importance
+
+We can use the most significant bit to represent the most important permission, the comparison between permissions can then be done easily since they all are `uint256`s.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+_Note_ The following specifications use syntax from Solidity `0.8.7` (or above)
```
[the remark this answers is not in the document]
Comment (chiro-hiro) [reply EIPS/eip-bit_based_permission.md]: @vdusart implemented a library to perform bit wise operators

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

File: EIPS/eip-6617.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        pure
+        returns (bool)
+    {
+        return _permission & _requiredPermission == _requiredPermission;
+    }
+
+    /**
+        @notice Add permission
+        @param _permission          The given permission
+        @param _permissionToAdd		The permission that will be added
```
Comment (chiro-hiro) [inline EIPS/eip-6617.md:119]: @vdusart I think, code formatter is broken.

**Review by Pandapip1 — APPROVED** [review]
Lgtm for draft

**Pull request merged**